src/server/ua_services_attribute.c

```diff
@@ -1150,13 +1158,15 @@ adjustValueType(UA_Server *server, UA_Variant *value,
     if(!value->type)
         return;
 
-    const UA_DataType *targetDataType = UA_findDataType(targetDataTypeId);
-    if(!targetDataType)
-        return;
-
     /* Unwrap ExtensionObject arrays if they all contain the same DataType */
     unwrapEOArray(server, value);
 
+    const UA_DataType *targetDataType = UA_findDataType(targetDataTypeId);
```

Comment:
We should use UA_findDataTypeWithCustom here.
So we can take the custom types loaded into the server into account.